ckanext/spectrum/actions.py

```diff
@@ -61,6 +64,70 @@ def user_create(next_action, context, data_dict):
     return created_user
 
 
+def dataset_tag_patch(context, data_dict):
+    if 'tags' not in data_dict or not isinstance(data_dict['tags'], dict):
+        raise toolkit.ValidationError(toolkit._(
+            "Must specify 'tags' dict of tags for update in form "
+            "{'old_tag_name1': 'new_tag_name1', 'old_tag_name2': 'new_tag_name2'}"))
```

Comment:
> We can do this in a "CKAN way" using [toolkit.get_or_bust](https://docs.ckan.org/en/2.9/extensions/plugins-toolkit.html#ckan.plugins.toolkit.ckan.plugins.toolkit.get_or_bust)

Good idea, I prefer CKAN way as well :).

Reply:
Unfortunately, due to bug in CKAN it can't be done this way.

When toolkit.get_or_bust("tags") is used an exception is being thrown. First, validation passes ok, i.e. one gets an error `{'tags': ['Missing value']}` . But, when `get_or_bust` creates `ValidationError` with those errors, there is a funny logic for `'tags'' in ckan/ckan/logic/__init__.py#74:

`# tags errors are a mess so let's clean them up
        if 'tags' in error_dict:
            tag_errors = []
            for error in error_dict['tags']:
                try:
                    tag_errors.append(', '.join(error['name']))
                except KeyError:
                    # e.g. if it is a vocabulary_id error
                    if error:
                        tag_errors.append(error)
            error_dict['tags'] = tag_errors`

Since `'tags'`  key in `error_dict` has no `'name'` attributes (it's a plain string) a `TypeErorr` is being thrown. So we need to stick with original version of checking for `'tags'`.